### PR TITLE
Add support for `GEOMETRY` type

### LINKED ISSUE #426: Add `GEOMETRY` support to snowflake-sqlalchemy
## What is the current behavior?
The snowflake sqlalchemy connector does not support the `GEOMETRY` data type.

## What is the desired behavior?
Support is added for the `GEOMETRY` data type.

## How would this improve `snowflake-connector-python`?
It would increase functionality for users of the snowflake-connector-python that use the `GEOMETRY` data type.

## References, Other Background
This would work would follow the pattern established [here](https://github.com/snowflakedb/snowflake-sqlalchemy/pull/271).

### PR DESCRIPTION
**Pull request by zachflanders**
Please answer these questions before submitting your pull requests. Thanks!

1. What GitHub issue is this PR addressing? Make sure that there is an accompanying issue to your PR.

   Fixes #426 

2. Fill out the following pre-review checklist:

   - [x] I am adding a new automated test(s) to verify correctness of my new code
   - [ ] I am adding new logging messages
   - [ ] I am adding new credentials
   - [ ] I am adding a new dependency

3. Please describe how your code solves the related issue.

   This code change adds support for the `GEOMETRY` data type.  It closely follows [this effort to add the `GEOGRAPHY` data type](https://github.com/snowflakedb/snowflake-sqlalchemy/pull/271).

### CODE REVIEW

**Comment by zachflanders** [comment]
It appears that the link to the Contributor License Agreement is broken.

**Comment by zachflanders** [comment]
I have read the CLA Document and I hereby sign the CLA

**Comment by zachflanders** [comment]
recheck

File: src/snowflake/sqlalchemy/__init__.py [hunk 1]
```diff
@@ -49,6 +49,7 @@
     DOUBLE,
     FIXED,
     GEOGRAPHY,
+    GEOMETRY,
```
Comment (sfc-gh-aling) [inline src/snowflake/sqlalchemy/__init__.py:52]: let's add the class in `__all__` as well

**Comment by sfc-gh-aling** [comment]
hi @zachflanders , thanks for the PR, it looks great to me.
I originally wanted to do some minor the PR however I don't find myself having the access to contribute to your fork.

so I created another one: https://github.com/snowflakedb/snowflake-sqlalchemy/pull/429

**Comment by sfc-gh-aling** [comment]
thanks for your contribution! I merged the other PR, it shall be carried in our next release

**Pull request closed without merging**